pkg/cmd/operator/cmd.go

```diff
@@ -67,35 +76,95 @@ func NewOperator() *cobra.Command {
 				log.WithError(err).Fatal("unable to set up client config")
 			}
 
-			// Create a new Cmd to provide shared dependencies and start components
-			log.Info("setting up manager")
-			mgr, err := manager.New(cfg, manager.Options{
-				MetricsBindAddress:      ":2112",
-				LeaderElection:          true,
-				LeaderElectionNamespace: minterv1.CloudCredOperatorNamespace,
-				LeaderElectionID:        leaderElectionConfigMap,
-			})
-			if err != nil {
-				log.WithError(err).Fatal("unable to set up overall controller manager")
+			run := func(ctx context.Context) {
+				// Create a new Cmd to provide shared dependencies and start components
+				log.Info("setting up manager")
+				mgr, err := manager.New(cfg, manager.Options{
+					MetricsBindAddress: ":2112",
+				})
+				if err != nil {
+					log.WithError(err).Fatal("unable to set up overall controller manager")
+				}
+
+				log.Info("registering components")
+
+				// Setup Scheme for all resources
+				util.SetupScheme(mgr.GetScheme())
+
+				// Setup all Controllers
+				log.Info("setting up controller")
+				kubeconfigCommandLinePath := cmd.PersistentFlags().Lookup("kubeconfig").Value.String()
+				if err := controller.AddToManager(mgr, kubeconfigCommandLinePath); err != nil {
+					log.WithError(err).Fatal("unable to register controllers to the manager")
+				}
+
+				// Start the Cmd
+				log.Info("starting the cmd")
+				if err := mgr.Start(signals.SetupSignalHandler()); err != nil {
+					log.WithError(err).Fatal("unable to run the manager")
+				}
 			}
 
-			log.Info("registering components")
-
-			// Setup Scheme for all resources
-			util.SetupScheme(mgr.GetScheme())
-
-			// Setup all Controllers
-			log.Info("setting up controller")
-			kubeconfigCommandLinePath := cmd.PersistentFlags().Lookup("kubeconfig").Value.String()
-			if err := controller.AddToManager(mgr, kubeconfigCommandLinePath); err != nil {
-				log.WithError(err).Fatal("unable to register controllers to the manager")
+			// Leader election code based on:
+			// https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/examples/leader-election/main.go#L130
```

Comment:
Make this a permalink, and cover the entire section that you are basing the code on.
https://github.com/kubernetes/kubernetes/blob/f7e3bcdec2e090b7361a61e21c20b3dbbb41b7f0/staging/src/k8s.io/client-go/examples/leader-election/main.go#L92-L154

Reply:
Fixed.